### ISSUE TITLE
Cant' return instance of custom class from a static function

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#35888]
**Godot version:**
3.2 stable

**Issue description:**
So uh, this is really weird, but I managed to get an error that says
![image](https://user-images.githubusercontent.com/2223172/73678471-be5e4f80-46b8-11ea-859f-9fe01e88eff8.png)

It happens when you try to return an object of a custom class from a singleton node in a static method of that class.

In my case, I have `Dialogue.gd` with Dialogue class and a singleton named Util. There's a node instance with Dialogue.gd script and I'm trying to return that instance, but from a static method inside the Dialogue class. It's convoluted, but I have reproduction project, so don't worry xd

Also it works when I do this
```
static func get_dialogue_node() -> Dialogue:
	return Util.game.UI.dialogue_panel as Dialogue
```
which makes it ten billion percent more weird.

**Minimal reproduction project:**
[TestProjectReproduction.zip](https://github.com/godotengine/godot/files/4149383/TestProjectReproduction.zip)

**Comment by staddy** [comment]
You are trying to return a member variable from static function, it doesn't make any sense. As I know, Godot doesn't support static member variables, but if you replace `var` with `const` to make it work.
So the bug is about incorrect error description.

**Comment by KoBeWi** [comment]
@staddy You totally misunderstood the issue. I have the instance and I can return it, but only after I cast it to the same type (or remove return type hint). The error is not only wrong, it shouldn't happen at all.

Just check the test project to see for yourself.

**Comment by KoBeWi** [comment]
Still valid in 3.2.4 beta4
In 2cbc0b6 the type isn't validated properly, so there's no parse error and when you try to return wrong type, it just silently returns null.

**Comment by tomwyr** [comment]
I stumbled upon similar issue in Godot v3.4.4.stable.official [419e713a2]
![image](https://user-images.githubusercontent.com/9600796/184517216-bece3df8-9faf-4213-86c0-1b1c0885a533.png)

**Comment by KoBeWi** [comment]
Fixed in 4.0
